test/base/strict-mode.js

```diff
@@ -241,6 +244,147 @@ test("warn on unsafe reads", function() {
+                x.yy
+            }, /being read outside a reactive context/)
+        } finally {
+            mobx.configure({ observableRequiresReaction: false })
+        }
+    })
+
+    test("warn on unsafe reads of observable also when there are other subscriptions", function() {
+        try {
+            mobx.configure({ observableRequiresReaction: true })
+            const x = mobx.observable({
+                y: 3
+            })
+
+            const dispose = mobx.autorun(() => x.y)
+
+            utils.consoleWarn(() => {
+                x.y
+            }, /being read outside a reactive context/)
+
+            dispose()
+        } finally {
+            mobx.configure({ observableRequiresReaction: false })
+        }
+    })
+
+    test("warn on unsafe reads of observable array", function() {
+        try {
+            mobx.configure({ observableRequiresReaction: true })
+            const x = mobx.observable({
+                arr: [1, 2, 3]
+            })
+            utils.consoleWarn(() => {
+                x.arr[1]
+            }, /being read outside a reactive context/)
+        } finally {
+            mobx.configure({ observableRequiresReaction: false })
+        }
+    })
+    test("don't warn on reads inside batch - computed", function() {
+        try {
+            mobx.configure({ observableRequiresReaction: true })
+            const x = mobx.observable({
+                y: 1
+            })
+
+            const fooComputed = mobx.computed(() => x.y + 1)
+
+            const messages = utils.supressConsole(() => {
+                const dispose = mobx.autorun(() => fooComputed.get())
+                dispose()
+            })
+
+            expect(messages.length).toBe(0)
+        } finally {
+            mobx.configure({ observableRequiresReaction: false })
+        }
+    })
+
+    test("don't warn on reads inside batch - action", function() {
+        try {
+            mobx.configure({ observableRequiresReaction: true })
+            const x = mobx.observable({
+                y: 1
+            })
+
+            const fooAction = mobx.action(() => x.y)
+
+            const messages = utils.supressConsole(() => {
+                fooAction()
+            })
+
+            expect(messages.length).toBe(0)
+        } finally {
+            mobx.configure({ observableRequiresReaction: false })
+        }
+    })
+
+    test("don't warn on reads inside batch - transaction", function() {
+        try {
+            mobx.configure({ observableRequiresReaction: true })
+            const x = mobx.observable({
+                y: 1
+            })
+
+            const messages = utils.supressConsole(() => {
+                mobx.transaction(() => x.y)
+            })
+
+            expect(messages.length).toBe(0)
+        } finally {
+            mobx.configure({ observableRequiresReaction: false })
+        }
+    })
+
+    test("don't warn on reads inside batch - untracked inside autorun", function() {
```

Comment:
`inside batch - untracked inside autorun` => `inside untracked` (not a batch)

Reply:
its also inside a batch, that why its not warning